fw/http_limits.c

```diff
@@ -194,6 +194,9 @@ frang_conn_new(struct sock *sk)
 	/* The new socket is allocated by inet_csk_clone_lock(). */
 	assert_spin_locked(&sk->sk_lock.slock);
 
+	if (tfw_http_mark_is_in_whitlist(skb->mark))
+		return T_OK;
```

Comment:
Don't we produce a warning on the HTTP request parsing callback `frang_http_req_handler()` due to uninitialized `sk->sk_security`?

Reply:
I don't understand a question. Yes we produce a warning. As I can see sk->security is initialized in `frang_conn_new` and it should not be zero in frang_http_req_process.

Reply:
If `tfw_http_mark_is_in_whitlist()` returns `true` (an IP is whitelisted), then `frang_conn_new()` returns immediately with `T_OK` and does not initialize `sk->sk_security`. This happens on a new connection, but when we get a request on this connection, then `frang_http_req_handler()` finds  `ra = sk->sk_security = NULL` and we get a warning.

I think we should set `sk->sk_securoty` to some specific value and check it in other Frang calls to avoid warnings/crashes on `NULL` value. Also we should run `bsearch()` on the while list just once (at it is now, not to introduce other searches).